Coral/Coral.js

```diff
@@ -0,0 +1,312 @@
+import { fix_mouse_coords } from "../common/fix_mouse_coords.js";
+
+const WIDTH = 500;
+const HEIGHT = 700;
+
+const QUAD_X = WIDTH / 2 - 100;
+const QUAD_Y = HEIGHT / 2 - 100;
+const QUAD_WIDTH = 200;
+const QUAD_HEIGHT = 200;
+
+function add(a, b) {
```

Comment:
Surprisingly, even with all the sketches I've made in this repository, I still don't have a vector class...

I'd like to implement 2D PGA, that would be helpful here and for other demos. I'll definitely need this when it comes to deforming the quad in time.